build/bazelutil/bazel-generate.sh

```diff
@@ -76,10 +76,13 @@ else
   bazel run pkg/gen/genbzl --run_under="cd $PWD && " -- --out-dir pkg/gen
 fi
 
+bazel build @com_github_bazelbuild_buildtools//buildozer:buildozer
+
 if files_unchanged_from_upstream $(find_relevant ./pkg/cmd/generate-test-suites -name BUILD.bazel -or -name '*.go') $(find_relevant ./pkg -name BUILD.bazel) $(find_relevant ./pkg -name '*.bzl'); then
   echo "Skipping //pkg/cmd/generate-test-suites (relevant files are unchanged from upstream)."
 else
-  bazel build @com_github_bazelbuild_buildtools//buildozer:buildozer
   CONTENTS=$(bazel run //pkg/cmd/generate-test-suites --run_under="cd $PWD && ")
   echo "$CONTENTS" > pkg/BUILD.bazel
 fi
+
+bazel run //pkg/cmd/generate-test-timeouts --run_under="cd $PWD && "
```

Comment:
You can put this in the branch above next to `generate-test-suites`, or in a new branch that checks `BUILD.bazel` files and `*.bzl` files. If none of the `BUILD.bazel` or `*.bzl` files have changed from `master` then `generate-test-timeouts` doesn't need to be run. 

Also we don't want to unconditionally build `buildozer`. Building small `go` executables is a non-negligible amount of latency, we don't want to force people to do this for no reason.

This is optional, but consider just merging what you currently have in `generate-test-timeouts` into `generate-test-suites`. `generate-test-suites` already depends on `buildozer` and does similar munging in the `BUILD.bazel` files, so the logic may just fit right in there.

Reply:
> This is optional, but consider just merging what you currently have in generate-test-timeouts into generate-test-suites. generate-test-suites already depends on buildozer and does similar munging in the BUILD.bazel files, so the logic may just fit right in there.

Right I adopted part of the code from `generate-test-suites` but I thought it might be better to separate them because `generate-test-suite` outputs something to stdout which is then forwarded to `pkg/BUILD.bazel`. Maybe I should make it write its output to `pkg/BUILD.bazel` directly (ie in the binary, instead of forwarding stdout to `pkg/BUILD.bazel`)? Then move the timeout code to the same package and rename it to `generate-test-suites-and-timeouts`?

Reply:
> Maybe I should make it write its output to pkg/BUILD.bazel directly (ie in the binary, instead of forwarding stdout to pkg/BUILD.bazel)? Then move the timeout code to the same package and rename it to generate-test-suites-and-timeouts?

This approach makes sense to me. For the new name `generate-test-suites-and-timeouts` is not ideal since the binary actually does a few different things at this point. Maybe `generate-bazel-extra` or something like that?

Reply:
Should we generate timeouts when `*.bzl` files change? I can't find a reason to do that but I don't know enough about everything we are doing `.bzl` files and I am wondering because we do update `pkg/BUILD.bazel` when a `*.bzl` file changes. My current understanding is that with timeouts we are only concerned about the test target size so if `BUILD.bazel` files didn't change and our binary `pkg/cmd/generate-bazel-extra` didn't change then we are good?

Reply:
Hmm. I'm not aware of any timeouts that would need to be updated when `.bzl` files change. But I think you do need to run `generate-test-suites` (especially the generation of `pkg/BUILD.bazel`) if `.bzl` files change.

Reply:
Right but I am thinking of doing this:

```
if ! (files_unchanged_from_upstream $(find_relevant ./pkg -name BUILD.bazel) $(find_relevant ./pkg/cmd/generate-bazel-extra -name BUILD.bazel -or -name '*.go')); then
  bazel build @com_github_bazelbuild_buildtools//buildozer:buildozer
  bazel run //pkg/cmd/generate-bazel-extra --run_under="cd $PWD && " -- -gen_test_suites -gen_tests_timeouts
elif files_unchanged_from_upstream $(find_relevant ./pkg -name '*.bzl'); then
  echo "Skipping //pkg/cmd/generate-bazel-extra (relevant files are unchanged from upstream)."
else
  echo "Skipping `generate tests timeouts` from //pkg/cmd/generate-bazel-extra (relevant files are unchanged from upstream)."
  bazel build @com_github_bazelbuild_buildtools//buildozer:buildozer
  bazel run //pkg/cmd/generate-bazel-extra --run_under="cd $PWD && " -- -gen_test_suites
```

so we can skip generating timeouts when we don't need to

Reply:
Seems sensible to me.